src/components/FreaksGrid/FreaksGrid.jsx

```diff
@@ -1,13 +1,32 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
 import FreakTile from '../FreakTile/FreakTile';
 import './FreaksGrid.scss';
+import { getFilteredFreaks } from '../../filters/freaksFilter';
 import { PhotoModelProps } from '../../models/freak';
 
 function FreaksGrid({ freaks }) {
-  const tiles = freaks.map((freak) => (
-    <FreakTile id={ freak.id } name={ freak.firstName } photo={ freak.photo } key={ freak.id } />
-  ));
+  const filters = useSelector((state) => state.filters);
+
+  const getTiles = () => {
+    const getFreakTiles = (freaks) => freaks.map((freak) => (
+      <FreakTile
+        id={ freak.id }
+        name={ freak.firstName }
+        photo={ freak.photo }
+        key={ freak.id }
+      />
+    ));
+
+    return getFreakTiles(getFilteredFreaks(freaks, filters));
+  };
+
+  const [ tiles, setTiles ] = useState(getTiles);
+
+  useEffect(() => {
+    setTiles(getTiles());
+  }, [ filters ]);
 
   return <div className="tiles">{ tiles }</div>;
```

Comment:
This is very unconventional way of rendering based on a specific prop.

you could just do as 

```suggestion
  return <div className="tiles">{ getFilteredFreaks(freaks, filters).map(renderFreakTile) }</div>;
```